servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/Identity.java

```diff
@@ -33,4 +35,14 @@ private Identity() {
     public static ContentCodec identity() {
         return IDENTITY;
     }
+
+    /**
+     * Returns {@code true} if the provided {@link ContentCodec codec} has name equal to {@code identity}.
+     *
+     * @param codec The {@link ContentCodec codec} to check.
+     * @return {@code true} if the provided {@link ContentCodec} has name equal to {@code identity}.
+     */
+    public static boolean isIdentity(@Nullable final ContentCodec codec) {
+        return IDENTITY.equals(codec);
```

Comment:
As per our discussion yesterday, we are keeping the equals solution that @hbelmiro provided, so this serves no other purposes than aesthetics. 
The initial reasoning was that if we did get rid of equals and enforced our own solution this would serve as the utility to verify the codec (having identity() == codec in so many places was not as nice as having a helper).
@hbelmiro feel free to revert and keep your initial approach, sorry for the back-forth here.

Reply:
It seems kind of strange to allow more than one Identity codec. It seems like it should be a singleton, built-in and nobody else should be able to create a codec which claims to be identity. What possible legitimate purpose is there for another identity codec? Adding contortions to recognize multiple identity codecs seems strange.

Reply:
There is difference between protocol specification, programming language features, and programming mistakes made by users. We can not be too restrictive. To be so, we have to make all API as final classes and never use interfaces. But it kills flexibility.

There is nothing wrong with having multiple implementations of something. Look at `HttpRequestMethod` as an example. Can users have multiple instances of `POST` method? Sure. Users may have a use-case when the default properties are not what they need. It's more important to follow the spec, and spec says that regardless of the properties, only the method name matters when you identify the method.

Same here. We have `ContentCodec` interface. We have no control over what is the implementation of that interface (even if `Identity` is a singleton, users can implemet interface). The only thing that http spec says is that the name defines the codec, and the comparison have to be [case-insensitive](https://tools.ietf.org/html/rfc7231#section-3.1.2.1). That's it. There is nothing wrong with having different `gzip` implementations with different compression level, or `identity` implementations with different [q values](https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.4) that we may support in future or different metrics behind `ContentCodec` (like measuring number of times the identity is used vs other codecs).

Can a user mess up things and create a non noop `ContentCodec` with "identity" name? Yes. Should we kill interfaces to prevent those types of programming mistakes? Probably not.